app/src/server/api/external/v1Api.router.ts

```diff
@@ -115,62 +103,89 @@ export const v1ApiRouter = createOpenApiRouter({
+        (await prisma.fineTune.findUnique({
+          where: { slug: modelSlug },
+        })) ?? undefined;
 
-      const typedFT = typedFineTune(fineTune);
-
-      if (typedFT.projectId !== key.projectId) {
+      if (inputPayload.model.startsWith("openpipe:") && !fineTune) {
         throw new TRPCError({
-          message: "The model does not belong to this project",
-          code: "FORBIDDEN",
+          message: "The model does not exist",
+          code: "NOT_FOUND",
         });
       }
 
-      try {
-        const completion = await getCompletion2(typedFT, inputPayload);
-        const inputTokens = completion.usage?.prompt_tokens ?? 0;
-        const outputTokens = completion.usage?.completion_tokens ?? 0;
+      let completion: ChatCompletion | Stream<ChatCompletionChunk>;
 
-        const cost = calculateCost(typedFT, 0, inputTokens, outputTokens);
+      if (fineTune) {
+        const typedFT = typedFineTune(fineTune);
 
-        // Don't `await` this to minimize latency
-        prisma.usageLog
-          .create({
-            data: {
-              fineTuneId: fineTune.id,
-              type: UsageType.EXTERNAL,
-              inputTokens,
-              outputTokens,
-              cost,
-            },
-          })
-          .catch((error) => captureException(error));
+        if (typedFT && typedFT.projectId !== key.projectId) {
+          throw new TRPCError({
+            message: "The model does not belong to this project",
+            code: "FORBIDDEN",
+          });
+        }
 
-        posthogServerClient?.capture({
-          distinctId: fineTune.projectId,
-          event: "fine-tune-usage",
-          properties: {
-            model: inputPayload.model,
-            inputTokens,
-            outputTokens,
-            cost,
-          },
-        });
+        try {
+          completion = await getCompletion2(typedFT, inputPayload);
+        } catch (error: unknown) {
+          console.error(error);
+          throw new TRPCError({
+            message: `Failed to get fine-tune completion: ${(error as Error).message}`,
+            code: "BAD_REQUEST",
+          });
+        }
+      } else {
+        try {
+          completion = await getOpenaiCompletion(key.projectId, {
+            ...inputPayload,
+            // @ts-expect-error gotta love overloads
+            stream: input.stream,
+          });
+        } catch (error: unknown) {
+          console.error(error);
+          throw new TRPCError({
+            message: `Failed to get OpenAI completion: ${(error as Error).message}`,
+            code: "BAD_REQUEST",
+            cause: error as Error,
+          });
+        }
+      }
 
-        return completion;
-      } catch (error: unknown) {
-        console.error(error);
-        throw new TRPCError({
-          message: `Failed to get completion: ${(error as Error).message}`,
-          code: "BAD_REQUEST",
+      const logRequest = ctx.headers["op-log-request"] === "true";
```

Comment:
I think we should default to reporting unless you explicitly opt out. That makes it much simpler for us to tell someone "if you want to start using openpipe with [your random openai wrapper library], just use our baseUrl" insetad of having to also tell them how to set a custom header with their wrapper lib, which may not even be possible.

Reply:
Sure. I think that should go in a separate PR as a larger refactor of our SDKs to allow streaming (currently our SDKs will break if you try to use the streaming features this PR enables for fine-tuned models).